src/main/java/org/jenkinsci/plugins/vSphereCloud.java

```diff
@@ -534,7 +533,7 @@ private static Node provisionNewNode(final CloudProvisioningRecord whatWeShouldS
             final vSphereCloudProvisionedSlave slave = template.provision(cloneName, StreamTaskListener.fromStdout());
             // ensure Jenkins knows about us before we forget what we're doing,
             // otherwise it'll just ask for more.
-            Jenkins.getInstance().addNode(slave);
+            Jenkins.get().addNode(slave);
```

Comment:
Everything is covered in javadoc, https://javadoc.jenkins-ci.org/jenkins/model/Jenkins.html#get-- in all of these cases it is .get when trying newer Jenkins.

Reply:
OK, I've read the javadocs and looked at the code...
It seems to me that the direct like-for-like replacement for `Jenkins.getInstance()` is `Jenkins.getInstanceOrNull()` and the direct like-for-like replacement for `Jenkins.getActiveInstance()` is `Jenkins.get()`.
However, in any situation in the code where it's currently calling `Jenkins.getInstance()` and where it is _not_ checking for a `null` result then that code is wrong and should really have been calling `Jenkins.getActiveInstance()` instead because that method handles the null situation much better than a generic NPE.
i.e. there's some tech-debt here.

The problem is that `Jenkins.get()` says it was introduced in Jenkins 2.98, and it wasn't back-ported to LTS, so the first LTS release of Jenkins that contained that was version 2.107.1 which was released on 14th March 2018 and that's not old enough for my liking - I think we should give folks at least a year to upgrade.

So, I think that we should hold off switching to using `Jenkins.get()` for now as that's a bit _too_ new ... but perhaps after we've updated to require Jenkins 2.60.3 then we could replace all calls to `Jenkins.getInstance()` with either `Jenkins.getInstanceOrNull()` or `Jenkins.getActiveInstance()` so that the code is 100% clear what it needs in each situation (and would make a later update to use `Jenkins.get()` much more straightforward).